core/src/main/java/feign/SynchronousMethodHandler.java

```diff
@@ -101,13 +103,20 @@ Object executeAndDecode(RequestTemplate template) throws Throwable {
     }
     long elapsedTime = TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - start);
 
+    boolean shouldClose = true;
     try {
       if (logLevel != Logger.Level.NONE) {
         response =
             logger.logAndRebufferResponse(metadata.configKey(), logLevel, response, elapsedTime);
       }
       if (Response.class == metadata.returnType()) {
-        if (response.body() == null) {
+        if (response.body() == null || response.body().isRepeatable()) {
+          return response;
+        }
+        if (response.body().length() == null ||
+                response.body().length() > MAX_RESPONSE_BUFFER_SIZE ||
+                response.body().length() < 0) {
```

Comment:
I think we should forbid this being -1, as it is redundant with the semantics of null. ex. enforce in InputStreamBody   ctor as -1 should be a bug. Have you ever seen this assigned to -1 in feign?